compiler/rustc_const_eval/src/transform/check_consts/resolver.rs

```diff
@@ -265,6 +273,19 @@ pub(super) struct State {
     pub borrow: BitSet<Local>,
 }
 
+impl Clone for State {
+    fn clone(&self) -> Self {
+        State { qualif: self.qualif.clone(), borrow: self.borrow.clone() }
+    }
+
+    // Data flow engine when possible uses `clone_from` for domain values. Implementing it should
+    // avoid some intermediate memory allocations.
+    fn clone_from(&mut self, other: &Self) {
```

Comment:
So this is not the default implementation generated by `derive`?

Reply:
Ah, that seems like an improvement worth making... but it has nothing to do with this PR, of course.

Reply:
Moved to a separate pull request.